ledfx/utilities/gradient_extraction.py

```
@@ -0,0 +1,989 @@
+                }
+            )
+
+        # Sort by frequency (most dominant first)
+        color_frequencies.sort(key=lambda x: x["frequency"], reverse=True)
+
+        # Deduplicate similar colors to prevent gradient dominated by close variants
+        deduplicated = _deduplicate_colors(color_frequencies)
+
+        return deduplicated
+
+    except Exception as e:
```

Ruff (0.15.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)